astrauli/user-posts-code-challenge
# Authentication routes and UI pages

### Description
- Add auth routes in the express server
- Add auth pages in the Nextjs project
- Add Redis to the existing `docker-compose.yml` file
  - This will be the data store for sessions
- Add passport to help with auth
  - Uses the local strategy for simplicity

### Testing
- Running `npm run dev` should start up the Redis instance and the server should automatically connect
- Go to the `/signup` page to verify signup routes return a successful response
- Go to the `/login` page to verify signup routes return a successful response (given the credentials used above)
  - In the dev console, you can see that a `connect.sid` value is populated in the `Set-Cookie` response header values. The session ID is saved in the cookie and can be utilized on subsequent requests.

<img width="1401" alt="Screenshot 2023-08-15 at 1 07 20 PM" src="https://github.com/astrauli/user-posts-code-challenge/assets/17238622/945ad3e6-5690-4f8a-83f0-f072be65be83">
